app/navigators/ChargeDNavigator.scala

```diff
@@ -114,14 +115,32 @@ class ChargeDNavigator @Inject()(val dataCacheConnector: UserAnswersCacheConnect
     case TaxQuarterReportedAndPaidPage(ChargeTypeLifetimeAllowance, index, schemeIndex) =>
       controllers.mccloud.routes.ChargeAmountReportedController
         .onPageLoad(ChargeTypeLifetimeAllowance, NormalMode, srn, startDate, accessType, version, index, schemeIndex)
-    case ChargeAmountReportedPage(ChargeTypeLifetimeAllowance, index, _) =>
-      CheckYourAnswersController.onPageLoad(srn, startDate, accessType, version, index)
+    case ChargeAmountReportedPage(ChargeTypeLifetimeAllowance, index, schemeIndex) =>
+      val schemeSize = countSchemeSize(ua, index)
```

Comment:
Can we move this into a method, so as to keep this large method concise?

Reply:
moved routing additional logic to private method